### PR TITLE
[Custom Amounts M2] Redesign Payments and Customers section

### LINKED ISSUE #9941: [Custom Amounts M2] Redesign Payments and Customers section
The objective of this task is to revamp both the Payments section and the Customers section.

The new design can be found here: ichqhxDjbmZZmvJaJkGhkj-fi-118%3A6643&mode=dev

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #9941 

### Description

This PR introduces a restructured Payment section within the order creation workflow, aligning it with our design specifications. ichqhxDjbmZZmvJaJkGhkj-fi-118%3A6643

### Testing instructions

1. Navigate to Order creation screen
3. Check that the new design matches:

- Empty State (no items added)
- Add products, custom amounts, shipping, coupons, discounts, and gift cards to check that it works fine.
- Add customer details and notes to check that it works according to design as well.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/1331230/5981d43c-cc88-4ebc-873c-a914118d2fc2

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 1]
```diff
@@ -907,11 +1048,18 @@ class OrderCreateEditFormFragment :
...
-            shippingButton.isEnabled = false
+            addShippingButton.isEnabled = false
             lockIcon.isVisible = true
             couponButton.isEnabled = false
             addCouponButton.isEnabled = false
         }
+        customAmountsSection.apply {
+            isLocked = true
+        }
+        val disabledCustomAmountUIModel = viewModel.customAmounts.value?.map {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: 💡 CustomAmountUIModel items are computed and emitted by view model. Fragment is doing too much logic here, and shouldn't be responsible for transforming their state.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 2]
```diff
@@ -907,11 +1048,18 @@ class OrderCreateEditFormFragment :
...
+            it.copy(isLocked = true)
+        }
+        modifyProductsAndCustomAmountsSectionUI(viewModel.viewStateData.liveData.value, this)
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]:
```suggestion
        updateProductsAndCustomAmountsSectionUI(viewModel.viewStateData.liveData.value, this)
```
💡 IMO, "modify" is a bit blurry and too broad verb. I struggle to understand what's the intent of the method by its name. How about "update..." or "render..."?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 3]
```diff
@@ -779,6 +822,96 @@ class OrderCreateEditFormFragment :
...
+                // Both customer address and customer notes are added
+                else -> {
+                    displayCustomerAddress(binding, newOrderData)
+                    displayCustomerNotes(binding, newOrderData)
+                }
+            }
+        }
+    }
+
+    private fun shouldHideBothSections(newOrderData: Order) =
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: 💡 Method's name is not meaningful enough. There are many sections in the screen.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 4]
```diff
@@ -779,6 +822,96 @@ class OrderCreateEditFormFragment :
...
+
+    private fun shouldShowCustomerSectionOnly(newOrderData: Order) =
+        (newOrderData.customer != null && newOrderData.customer != Order.Customer.EMPTY) &&
+            newOrderData.customerNote.isEmpty()
+
+    private fun shouldShowNotesSectionOnly(newOrderData: Order) =
+        (newOrderData.customer == null || newOrderData.customer == Order.Customer.EMPTY) &&
+            newOrderData.customerNote.isNotNullOrEmpty()
+
+    private fun FragmentOrderCreateEditFormBinding.hideBothSections() {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: 💡 Method's name is not meaningful enough. It's not clear what sections are going to be hidden by reading its name.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 5]
```diff
@@ -779,6 +822,96 @@ class OrderCreateEditFormFragment :
...
+            customerSection.apply {
+                show()
+                hideHeader()
+            }
+            displayCustomerNotes(this, newOrderData)
+            initCustomerSection()
+        }
+    }
+
+    private fun displayCustomerNotes(binding: FragmentOrderCreateEditFormBinding, newOrderData: Order) {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: 💡 Can we keep this one consistent with the above functions and refactor it to be an extension function?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 6]
```diff
@@ -779,6 +822,96 @@ class OrderCreateEditFormFragment :
...
+            initNotesSection()
+            notesSection.apply {
+                show()
+                showHeader()
+                bindNotesSection(this, newOrderData.customerNote)
+            }
+        }
+    }
+
+    private fun displayCustomerAddress(binding: FragmentOrderCreateEditFormBinding, newOrderData: Order) {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: 💡 Can we keep this one consistent with the above functions and refactor it to be an extension function?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt [hunk 7]
```diff
@@ -1400,7 +1403,8 @@ data class ProductUIModel(
 data class CustomAmountUIModel(
     val id: Long,
     val amount: BigDecimal,
-    val name: String
+    val name: String,
+    val isLocked: Boolean = false
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt:1418]: 💡 `isLocked` value mirrors `viewState.isEditable`. This suggests the observable emitting that type should be updated whenever `viewState.isEditable` changes or that `isLocked` property is redundant.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 8]
```diff
@@ -334,12 +354,8 @@ class OrderCreateEditFormFragment :
 
     @Suppress("LongMethod")
     private fun observeViewStateChanges(binding: FragmentOrderCreateEditFormBinding) {
-        if (isCustomAmountsFeatureFlagEnabled()) {
-            viewModel.combinedProductAndCustomAmountsLiveData.observe(viewLifecycleOwner) {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt:338]: ❓ The `combinedProductAndCustomAmountsLiveData` property is now not used anywhere

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 9]
```diff
@@ -459,13 +490,18 @@ class OrderCreateEditFormFragment :
...
+        if (viewModel.viewStateData.liveData.value?.isEditable == true) {
+            binding.customAmountsSection.showAddAction()
+            binding.productsSection.showAddProductsHeaderActions()
+        } else {
+            binding.customAmountsSection.hideAddAction()
+            binding.productsSection.hideAddProductsHeaderActions()
+        }
     }
 
     private fun bothProductsAndCustomAmountsAreUnset(binding: FragmentOrderCreateEditFormBinding) {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt:512]: 💡 I know this is not from the current PR, but this function name is also missing a verb describing its intent, e.g. "render..."

**Review by samiuelson — COMMENTED** [review]
I skimmed through the code and added a few improvement suggestions. 
I'll spend some more time testing the app on a device tomorrow.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Good point. Moved here https://github.com/woocommerce/woocommerce-android/pull/10122/commits/3fb29f87a6cbd3f4ab45d5c79b7f542b62f5bdd8

Please note that I'll add unit tests for these in a separate PR.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 2 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Done https://github.com/woocommerce/woocommerce-android/pull/10122/commits/f31148d4f5aa27755245a7e0b90dbb593129ec0e

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 3 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Renamed https://github.com/woocommerce/woocommerce-android/pull/10122/commits/e729903c2632bd1ba517dcfa512bfea727dfa34d

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 5 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Done https://github.com/woocommerce/woocommerce-android/pull/10122/commits/d9bbbff43201119cfed85c57e5a2f259c3638443

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 6 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Done https://github.com/woocommerce/woocommerce-android/pull/10122/commits/75c476aa2392bf61481c8752aad1a9189bf27f9e

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 8 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt:338]: Good catch. Merge conflict error. Fixed it

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 9 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt:512]: I see a few more methods with similar naming conventions. I'll take those in a separate PR.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt (hunk 7 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt:1418]: I need some time to think about this. I'll take this as part of M3 which goes out next week if that's okay with you.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 4 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Done https://github.com/woocommerce/woocommerce-android/pull/10122/commits/d9bbbff43201119cfed85c57e5a2f259c3638443

**Comment by AnirudhBhat** [comment]
@samiuelson Thank you for the feedback! Appreciate it 🙏 I have updated the PR addressing those and a couple of them I plan to take it in a separate PR.

**Review by samiuelson — APPROVED** [review]
##### Overall, the UI updates look good. There are two simple things that could be done as well to match the designs:
1. Changing order status "edit" button to "pen" icon
2. Renaming "Payment" title to "Payment totals"


##### Other than that, not related to this PR but important, I noticed that the custom amount total's input is not behaving correctly:

https://github.com/woocommerce/woocommerce-android/assets/4527432/fcb1f633-9554-48b5-8c92-e356507092f6

💡 It would be great to align it with the behavior in the simple payments flow.

WDUT?

**Comment by AnirudhBhat** [comment]
Thanks for the review @samiuelson 🙏 

> Overall, the UI updates look good. There are two simple things that could be done as well to match the designs:
Changing order status "edit" button to "pen" icon
Renaming "Payment" title to "Payment totals"

These are done. Thank you!

> Other than that, not related to this PR but important, I noticed that the custom amount total's input is not behaving correctly:

Yes. This is a known issue. I'm working on it. It will come in a separate PR.

**Pull request merged**